cli.sh

```diff
@@ -1493,60 +1493,136 @@ call_script() {
 }
 
 #####################################################
-# Downloads a file "$1" with `wget` or `curl`
+# Does a release clone repository exist in the
+#  install folder
+#
+# Returns: 0 if `true`, 1 otherwise
+#####################################################
+is_release_clone_existing() {
+    if git -C "$INSTALL_DIR/release" rev-parse >/dev/null 2>&1; then
+        return 0
+    fi
+    return 1
+}
+
+#####################################################
+# Loads the contents of the latest install
+#   script into a file ${INSTALL_SCRIPT}.
+#
+# Sets the ${INSTALL_SCRIPT} variable
 #
 # Returns:
-#   0 if download succeeded, 1 otherwise
+#   1 if failed to load the script, 0 otherwise
 #####################################################
-download_file() {
-    DOWNLOAD_FILE="$1"
-    OUTPUT_FILE="$2"
-    DOWNLOAD_APP=$(get_tool_script "download")
+fetch_latest_install_script() {
+    update_release_clone || return 1
+    return 0
+}
 
-    if [ "$DOWNLOAD_APP" != "" ]; then
-        echo "  Using App at \`$DOWNLOAD_APP\`"
-        # Use the external download app for downloading the file
-        call_script "$DOWNLOAD_APP" "$DOWNLOAD_FILE" "$OUTPUT_FILE"
-    else
-        # The main update url.
-        MAIN_DOWNLOAD_URL="https://raw.githubusercontent.com/rycus86/githooks/master"
+############################################################
+# Checks whether the given directory
+#   is a Git repository (bare included) or not.
+#
+# Returns:
+#   1 if failed, 0 otherwise
+############################################################
+is_git_repo() {
+    git -C "$1" rev-parse >/dev/null 2>&1 || return 1
+}
 
-        # Default implementation
-        DOWNLOAD_URL="$MAIN_DOWNLOAD_URL/$DOWNLOAD_FILE"
+#####################################################
+# Updates the release clone in the install folder
+#
+# Sets the ${INSTALL_SCRIPT} variable
+#
+# Returns:
+#   1 if failed, 0 otherwise
+#####################################################
+update_release_clone() {
 
-        echo "  Download $DOWNLOAD_URL ..."
-        if curl --version >/dev/null 2>&1; then
-            curl -fsSL "$DOWNLOAD_URL" -o "$OUTPUT_FILE" >/dev/null 2>&1
-        elif wget --version >/dev/null 2>&1; then
-            wget -O "$OUTPUT_FILE" "$DOWNLOAD_URL" >/dev/null 2>&1
-        else
-            echo "! Cannot download file \`$DOWNLOAD_URL\` - needs either curl or wget"
-            return 1
-        fi
+    if ! is_git_repo "$INSTALL_DIR/release"; then
+        clone_release_repository || return 1
     fi
 
-    # shellcheck disable=SC2181
+    #shellcheck disable=2154
```

Comment:
```suggestion
    # shellcheck disable=SC2154
```

Why is this necessary btw? We already use `$INSTALL_DIR` above.

Reply:
Removed.

Reply:
Removed.